walkdir.py

```diff
@@ -202,11 +202,32 @@ def onloop(dirpath):
                     continue
         yield dir_entry
 
+
+def handle_dirsymlink_as_file(walk_iter):
+    """Handle symlink to directories as files during the walk.
+
+       By default os.walk insert symlink to directories to the subdirs list.
+       This function move symlink to directories from the subdirs list to the file list.
+
+       NOTE: This function is not needed when using the followlinks option.
+       when following links the dirsymlinks will be walked when entering the linked dir.
```

Comment:
This warning can be a bit stronger, as this filter effectively *overrides* the followlinks setting on the underlying walk for the default case of a top down walk:

    NOTE: For top down walks, this filter will override the *followlinks* setting on the underlying walk, as `os.walk`
    and `os.fwalk` will only descend into directories that remain in the list of subdirectories after the parent
    directory is processed. For bottom up walks, symlinked directories will still be visited when *followlinks* is set,
    but applying this filter will make them appear as files when the parent directory is processed.

We'd then need a couple of additional test cases to make sure both of those statements are true (and stay that way as other changes are made).